client/src/views/functionality-cartography.vue

```diff
@@ -1136,6 +1138,13 @@
         if (node.children) {
           for (let i in node.children) {
             const child = node.children[i]
+            if (child.row.type === 'FUNCTIONALITY') {
+              if (node.isSelected && !child.isSelected) {
+                this.counts.selected += 1
+              } else if (!node.isSelected && child.isSelected) {
+                this.counts.selected -= 1
+              }
+            }
             child.isSelected = node.isSelected
```

Comment:
wouldn't be simpler to write something like below?

```
if (child.row.type === 'FUNCTIONALITY' && node.isSelected !== child.isSelected) {
  const increment = node.isSelected ? 1 : - 1
  this.counts.selected += 1
}
```

I didn't understand the intent of your code right away. Feel free to tell me what you think ;)